File: Mobile/weatherapp/services/city.service.js
export class CityService {
  url;
  constructor() {
    this.url = "https://eb37aa91d52e.ngrok.io/api/city";
    //"https://localhost:5001/api/city"
  }

  async getAllCities() {
    const request = await fetch(this.url);
    return await request.json();
  }

  async getCityById(id) {
    const request = await fetch(this.url + "/" + id);
    return await request.json();
  }

  async getClientCity() {
    const request = await fetch(this.url + "/client");
    return await request.text();
  }

  async getCitiesByIds(ids) {
    const request = await fetch(this.url + "/ids/" + ids);
    return await request.json();
  }
}

File: Backend/WeatherApp/WeatherApp/Services/CityService.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Net;
using System.Net.NetworkInformation;
using System.Net.Sockets;
using System.Threading.Tasks;
using IpData;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Options;
using Newtonsoft.Json;
using WeatherApp.Database;
using WeatherApp.Interfaces;
using WeatherApp.Models;

namespace WeatherApp.Services
{
    public class CityService : ICityService
    {
        private readonly ApplicationContext _context;
        private readonly ApplicationSettings _appSettings;

        public CityService(ApplicationContext context, IOptions<ApplicationSettings> appSettings)
        {
            _context = context;
            _appSettings = appSettings.Value;
        }

        public async Task<List<City>> GetAllCities()
        {
            return await _context.Cities.ToListAsync();
        }

        public async Task<City> GetCityById(int id)
        {
            return await _context.Cities.FindAsync(id);
        }

        public async Task<string> GetClientsCity(string remoteIpAddress)
        {
            var client = new IpDataClient(_appSettings.IpDataApiKey);
            var ipInfo = await client.Lookup(remoteIpAddress);

            return ipInfo.City;
        }

        public async Task<List<City>> GetCitiesByIds(string ids)
        {
            var idsList = ids.Split("_");
            var result = new List<City>();

            for(int i = 0; i < idsList.Length; i++) 
            {
                if(idsList[i] != string.Empty)
                {
                    result.Add(await _context.Cities.FindAsync(Convert.ToInt32(idsList[i])));
                }
            }

            return result;
        }
    }
}
